### PR TITLE
Add initial schema and automatic interface generation

### PR DESCRIPTION
**Pull request by bruno-f-cruz**
This PR adds a first schema version for the RFID reader device (device.yml) and the necessary .Net generators to build the high-level bonsai interface.

### CODE REVIEW

File: device.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    address: 43
+    type: U8
+    access: Read
+    visibility: private
+    description: Reserved for future use.
+  Reserved1:
+    << : *reserved
+    address: 44
+    description: Reserved for future use.
+  MatchTagId0: &matchtagid #TODO not sure what event this triggers...
```
Comment (bruno-f-cruz) [inline device.yml:77]: @filcarv It is not clear what event is triggered here since this event is write only...

File: device.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    << : *matchidduration
+    address: 53
+    description: The time the digital output pin will stay on (ms) if any tag is detected.
+  PulseDO0Duration:
+    address: 54
+    type: U16
+    access: Write
+    description: Triggers the digital output pin for the specified duration (ms).
+bitMasks:
+  HardwareNotifications: #TODO SOMETHING IS WRONG HERE
```
Comment (bruno-f-cruz) [inline device.yml]: Is this a typo in the firmware or do all the outputs have the same mask?

Comment (bruno-f-cruz) [reply device.yml]: Following the fix in #2 . These masks should be changed to:
      Buzzer: 0x1
      TopLed: 0x2
      BottomLed: 0x4

File: device.yml (hunk 1 above)
Comment (glopesdev) [reply device.yml:77]: @bruno-f-cruz I was just discussing this with @filcarv and these registers are actually used as filters to allow only specific IDs to trigger the `InboundDetectionId` event, i.e. if any of these registers is written, then only IDs in the list will trigger the event. By default, all IDs are allowed.

**Review by glopesdev — APPROVED** [review]
Overall it looks good, we need to update the generators to the latest version, but it might make more sense to do that in a new PR since this one is already quite complex.

**Pull request merged**